app/models/collection_store.rb

```diff
@@ -33,6 +33,14 @@ def recursive_collection_path(collection, path, starting_collection)
       starting_collection[:paths] << path
     else
       collection[:collections].each do |parent_collection|
+        if parent_collection == collection[:id]
+          puts "SELF CONTAINED COLLECTION: #{collection[:id]}"
+          next
+        elsif @data[parent_collection][:collections].include?(collection[:id])
+          puts "HAS JOINT COLLECTION: #{collection[:id]}"
+          next
+        end
+
```

Comment:
I will take a look tomorrow, but we should easily be able to just check if there is a cycle and stop the current recursion there, but continue the overall building of paths. It wouldn't even require owners to make changes to their collections.

Reply:
You missed an `end` after your `next` statement, but otherwise I think this is the way.

Reply:
``` ruby
if collection_already_in_path?(parent_id, path) 
  # we've reached a cycle: don't recurse further but save the path 
  starting_collection[:paths] << path
```

We actually don't want to save the path, yet. If the loop continues, the path will get messy. We only save the path at the end of the loop. 

As I noted, we should correct the collection's relationship anyway: remove the self containing and jointly connected relationships.

Reply:
Can you expand on what you mean by the path getting messy? There are multiple paths being created.

Reply:
After some local testing, I could not find any issues with adding the path if the collection is already there. Thanks @fenekku